pkg/publicapi/client/v2/client.go

```diff
@@ -55,12 +55,28 @@ type httpClient struct {
 // and deserialize the response into a response object
 func (c *httpClient) Get(ctx context.Context, endpoint string, in apimodels.GetRequest, out apimodels.GetResponse) error {
 	r := in.ToHTTPRequest()
-	_, resp, err := requireOK(c.doRequest(ctx, http.MethodGet, endpoint, r)) //nolint:bodyclose // this is being closed
-	if err != nil && resp != nil && resp.StatusCode == http.StatusUnauthorized {
-		return apimodels.ErrInvalidToken
-	} else if err != nil {
+
+	_, resp, err := c.doRequest(ctx, http.MethodGet, endpoint, r) //nolint:bodyclose // this is being closed
+	if err != nil {
 		return err
 	}
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		return apimodels.NewUnauthorizedError("invalid token")
+	}
```

Comment:
Do you think it would be cleaner to do:
```go
return apimodels.NewAPIError(http.StatusUnauthorized, "invalid token")
```
instead?

The methods like `NewBadRequestError`, `NewForbiddenError`, etc. feel like unnecessary wrapping.